test/pytest/test_keras_api.py

```diff
@@ -0,0 +1,483 @@
+                     use_bias=False,
+                     data_format='channels_last'))
+    model.add(Activation(activation='relu'))
+
+    model.compile(optimizer='adam', loss='mse')
+    X_input = np.random.rand(10,128,4)
+    keras_prediction = model.predict(X_input)
+    config = hls4ml.utils.config_from_keras_model(model)
+    hls_model = hls4ml.converters.convert_from_keras_model(model, hls_config=config)
+    hls_model.compile()
+    hls_prediction = hls_model.predict(X_input)
+
+    assert len(model.layers) + 2 == len(hls_model.get_layers())
+    assert list(hls_model.get_layers())[1].attributes['name'] == model.layers[0]._name
+
+    print(list(hls_model.get_layers())[1].attributes)
+    if conv1d == 'Conv1D':
+      assert list(hls_model.get_layers())[1].attributes['class_name'] == 'Conv1D'
+    assert list(hls_model.get_layers())[1].attributes['activation'] == str(model.layers[0].activation).split()[1]
+    assert list(hls_model.get_layers())[1].attributes["in_width"] == model.layers[0]._batch_input_shape[1]
+    assert list(hls_model.get_layers())[1].attributes['filt_width'] == model.layers[0].kernel_size[0]
+    assert list(hls_model.get_layers())[1].attributes['n_chan'] == model.layers[0].input_shape[2]
+    assert list(hls_model.get_layers())[1].attributes['n_filt'] == model.layers[0].filters
+    assert list(hls_model.get_layers())[1].attributes['stride_width'] == model.layers[0].strides[0]
+    assert list(hls_model.get_layers())[1].attributes['padding'] == model.layers[0].padding
+    assert list(hls_model.get_layers())[1].attributes['data_format'] == model.layers[0].data_format
+    assert list(hls_model.get_layers())[1].attributes["out_width"] == list(model.layers[0].output_shape)[1]
+
+    out_width	= math.ceil(float(model.layers[0]._batch_input_shape[2]) / float(model.layers[0].strides[0]))
+    pad_along_width	= max((out_width - 1) * model.layers[0].strides[0] + model.layers[0].kernel_size[0] - model.layers[0]._batch_input_shape[2], 0)
+    pad_left = pad_along_width // 2
+    pad_right = pad_along_width - pad_left
+
+    out_valid	= math.ceil(float(model.layers[0]._batch_input_shape[1] - model.layers[0].kernel_size[0] + 1) / float(model.layers[0].strides[0]))
+
+    if model.layers[0].padding == 'same':
+        assert list(hls_model.get_layers())[1].attributes['pad_left'] == pad_left
+        assert list(hls_model.get_layers())[1].attributes['pad_right'] == pad_right
+
+    elif model.layers[0].padding == 'valid':
+        assert list(hls_model.get_layers())[1].attributes['pad_left'] == 0
+        assert list(hls_model.get_layers())[1].attributes['pad_right'] == 0
+
+
+# # DONE
+# keras_conv2d = [Conv2D]
+# padds_options = ['same', 'valid']
+# chans_options = ['channels_first', 'channels_last']
+# @pytest.mark.parametrize("conv2d", keras_conv2d)
+# @pytest.mark.parametrize("chans", chans_options)
+# @pytest.mark.parametrize("padds", padds_options)
+# def test_conv2d(conv2d, chans, padds):
```

Comment:
I've checked, and apart from the tests with `channels_first`, this works and seems useful. So I've uncommented all that and it's now included.

I did a bit more tidy up of this file as well, since the three different activation test functions could be combined.